support-frontend/scripts/ssr.js

```diff
@@ -0,0 +1,87 @@
+const { writeFileSync } = require('fs');
+const { resolve } = require('path');
+const jsdom = require('jsdom');
+const { JSDOM } = jsdom;
+const { window } = new JSDOM(`...`, { url: 'http://localhost/' });
+
+//------------- Set up global state --------------
+
+global.URL = require('url').URL;
+global.URLSearchParams = require('url').URLSearchParams;
+
+global.window = window;
+global.window.guardian = {
+  settings: {},
+};
+
+global.localStorage = {
+  getItem: () => '',
+  setItem: (k) => {},
+};
+global.sessionStorage = global.localStorage;
+global.document = window.document;
+global.navigator = {
+  userAgent: 'node.js',
+};
+
+global.Image = function () {return this;};
+
+
+function writePage(path, html) {
+  console.log(`Writing ${path}`);
+
+  writeFileSync(
+    resolve(__dirname, '..', 'conf/ssr-cache/', `${path.replace(/\//g, '-')}.html`),
+    html, 'utf8');
+  console.log(`Finished writing ${path}`);
+}
+
+
+//------------- Showcase page --------------
+
+const { getHtml: getShowcase } = require('../public/compiled-assets/javascripts/showcasePage');
+
+writePage('uk/support', getShowcase());
+
+//------------- Paper landing page --------------
+
+window.guardian.productPrices = {
```

Comment:
not a huge fan of this for launch, they are nullable in the latest master so they can be empty and I think that's cleaner than hardcoding them